docs/user_guides/single_table/custom_constraints.rst

```diff
@@ -0,0 +1,263 @@
+logic on how you would like to apply it to your data. There are three 
+main functions that you can create:
+
+- ``transform`` which is responsible for the forward pass when using ``transform`` strategy.
+- ``reverse_transform`` which defines how to reverse the transformation of the ``transform``.
+- ``is_valid`` which indicates which rows satisfy the constraint and which ones do not.
+
+Let's look at a demo dataset:
+
+.. ipython:: python
+    :okwarning:
+
+    from sdv.demo import load_tabular_demo
+
+    employees = load_tabular_demo()
+    employees
+
+The dataset defined in :ref:`_single_table_constraints` contains basic details about employees.
+We will use this dataset to demonstrate how you can create your own constraint. 
+
+
+Using the ``CustomConstraint``
+------------------------------
+
+Let's consider the following example where we wish to generate synthetic data and we want
+a particular column, ``salary`` for example, to be a multiple of a *base* value, e.g. 500. 
+In other words, the ``salary`` increments by 500. We will define ``transform`` and 
+``reverse_transform`` methods to make sure our data satisfy our constraints.
+
+We can achieve our goal by performing transformations in a 2 step process:
+
+- Divide ``salary`` by the base unit (500). This transformation makes it easier to model the data
+  since it would now learn regular integer values without any explicit constraint on the data.
+- Reversing the effect by multiplying ``salary`` back with the base unit. Now that the model has 
+  learned regular integer values, we multiply it with the base (500) such that it now conforms to our original data range.
+
+
+.. ipython:: python
+    :okwarning:
+
+    def transform(table_data):
+        base = 500
+        table_data['salary'] = table_data['salary'] / base
+        return table_data
+
+
+After defining ``transform`` we create ``reverse_transform`` that reverses the operations made.
+
+.. ipython:: python
+    :okwarning:
+
+    def reverse_transform(table_data):
+        base = 500
+        table_data['salary'] = table_data['salary'].round() * base
+        return table_data
+
+
+Then, we pack every thing together in ``CustomConstraint`` and we can create our model 
+and generate synthetic data by passing the constraint we just created.
+
+.. ipython:: python
+    :okwarning:
+
+    constraint = CustomConstraint(
+        transform=transform, 
+        reverse_transform=reverse_transform
+    )
+
+    gc = GaussianCopula(constraints=[constraint])
+
+    gc.fit(employees)
+
+    sampled = gc.sample(10)
+
+When we view the ``sampled`` data, we should find that all the rows in the sampled 
+data have a salary that is a multiple of the base value.
+
+.. ipython:: python
+    :okwarning:
+
+    sampled
+
+
+Can I apply the same function to multiple columns?
+--------------------------------------------------
+
+Say we want ``salary`` and ``annual_bonus`` to be both composed of the base unit. Rather than 
```

Comment:
Following the same style above, I would not phrase this as a hypothesis, but rather as an observation.

"In the example above we fixed the ``salary`` format, but if we continue observing the data we will see that ``annual_bonus`` is also constrained by the same logic".